cachematrix.R

```diff
@@ -1,15 +1,35 @@
-## Put comments here that give an overall description of what your
-## functions do
+## These functions generate, cache, and return the cache of the inverse
+## of a matrix.
 
-## Write a short comment describing this function
+## This function creates a "matrix" object and caches its inverse.
 
 makeCacheMatrix <- function(x = matrix()) {
-
+  m <- NULL
+  set <- function(y) {
+    x <<- y
+    m <<- NULL
+  }
+  get <- function() x
+  setsolve <- function(solve) m <<- solve
+  getsolve <- function() m
+  list(set = set, get = get,
+       setsolve = setsolve,
+       getsolve = getsolve)
 }
 
 
-## Write a short comment describing this function
+## This function computes the inverse of the matrix returned by 
+## makeCacheMatrix (the previous function).  If the inverse has already
```

Comment:
It would seem (from the code) that the expected input is of the format returned by the previous function definition, but the way your describing it makes it seems as if the function is directly connected to the previous function (which is not the case).  It is probably worth spelling what the expected input is and its format, rather than vaguely referencing the previous function.